### PR TITLE
Add docker-compose as dev env and remove transformers

### LINKED ISSUE #88: [feat]: Streamer API testing
We will add streamer for backend APIs. We need to make sure frontend can support backend APIs.

### PR DESCRIPTION
**Pull request by Aisuko**
**Description**

This PR fixes #88  fixes: #44 

**Notes for Reviewers**

This PR also fixed the CI issue. And it also include container environment for development

## Note:
 It is hard to create all containers inside CI. So, we remove CI first, and add some mock API for FastAPI. This can bring many risks.

Now, we support development with docker-compose dev container. It will help us load all dependencies containers. And we can launch framework from vscode launch.json.

**[Signed commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by Micost — APPROVED** [review]

File: backend/src/repository/ai_models.py [hunk 1]
```diff
@@ -1,62 +1,104 @@
...
+    - to compute embeddings by using the encoder model
+    - to generate an answer by using the transformers pipeline.
+    """
+
     def __init__(self, model_name=DEFAULT_MODEL):
-        print("start init model")
-        self.pipeline, self.tokenizer = self.initialize_model(DEFAULT_MODEL)
-        self.initialize_encoder(DEFAULT_ENCODER)
+        #TODO Logger system
+        self.pipe, self.tokenizer = self.initialize_model(DEFAULT_MODEL)
```
[the remark this answers is not in the document]
Comment (Aisuko) [reply backend/src/repository/ai_models.py]: Nice catch. Done

**Comment by Aisuko** [comment]
Let's waiting for CI result.

File: backend/src/config/settings/base.py [hunk 2]
```diff
@@ -73,10 +73,13 @@ class BackendBaseSettings(BaseSettings):
...
     JWT_ALGORITHM: str = decouple.config("JWT_ALGORITHM", cast=str)  # type: ignore
 
-    class Config(pydantic.BaseConfig):
+    class Config(pydantic.ConfigDict):
         case_sensitive: bool = True
         env_file: str = f"{str(ROOT_DIR)}/.env"
         validate_assignment: bool = True
+        # https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.extra
+        #TODO: We need to make sure pydanic is really useful
+        extra='allow'
```
Comment (Aisuko) [inline backend/src/config/settings/base.py:82]: allow extra

Comment (Micost) [reply backend/src/config/settings/base.py:82]: that should also remove the validation errors

**Review by Micost — APPROVED** [review]

**Pull request merged**